readthedocs/doc_builder/environments.py

```diff
@@ -456,7 +490,7 @@ def run_command_class(
 
         # ``build_env`` is passed as ``kwargs`` when it's called from a
         # ``*BuildEnvironment``
-        build_cmd = cls(cmd, **kwargs)
+        build_cmd = cls(cmd, build_env=self, **kwargs)
```

Comment:
This is not safe. We should add `build_env` to kwargs instead.

Example:

```
>>> def fn(x, y, z):
...     return f"{x}, {y}, {z}"
... 
>>> kwargs = {"x": 1, "y": 2, "z": 3}
>>> fn(**kwargs)
'1, 2, 3'
>>> fn(y=2, **kwargs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __main__.fn() got multiple values for keyword argument 'y'
```

Reply:
build_env isn't passed from anywhere in our code, but changed. Also, we should probably just refactor this to declare the specific arguments instead of `**kwargs`, but leaving that for another day...